chesster.js

```diff
@@ -1169,3 +1177,71 @@ controller.hears([
         }
     });
 });
+
+
+/* Scheduling */
+
+// reply to any incoming message
+controller.on('ambient', function(bot, message) {
+    bot_exception_handler(bot, message, function(){
+        var channel = channels.byId[message.channel];
+        var scheduling_options = config.scheduling[channel.name];
+        if (!scheduling_options) {
+            return;
+        } 
+        try {
+            var results = spreadsheets.parse_scheduling(message.text, {
+                offset_hours: scheduling_options.offset_hours
+            });
+            var white = users.getByNameOrID(results.white);
```

Comment:
I know you said to ignore this from below, but the other reason we do this, is that when the @ version of names are used, we don't get their username. We get their id. Literally if I say:

@endraws vs @lakinwecker

the message looks like:
@<Uds7sD)7fHS> vs @<UHSDHF728>  Where the angle brackets surround a user id